### ISSUE TITLE
Multiclass segmentation

### ISSUE THREAD

**Issue by Hampusli** [catalyst-team/catalyst#1080]
## ❓ Questions and Help

Hi, I can't seem to find any documentation or examples for how multiclass segmentation would work. I have tried to rewrite the binary classification example for my solution without success and would be grateful for some assistance. 

I have 4 classes with masks where each pixel is set to 1-4 depending on what class it is. How should I rewrite the __getitem__ function for my problem?

PS
- [x] I know, that I could [join Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion.

**Comment by Atharva-Phatak** [comment]
Maybe this Kaggle kernel will help, it uses catalyst for the most part of the training and evaluation. 

Link:  https://www.kaggle.com/artgor/segmentation-in-pytorch-using-convenient-tools

**Comment by Hampusli** [comment]
Thank you for the help, it gave some assistance, but my problem still persists.

This is a link to a copy of my code: https://colab.research.google.com/drive/1I88d-9gLkbfaE4m8B8UtKAzx-eARo2kI?usp=sharing

I am unsure of what the input to the model should be as I get a "shape mismatch" when I try to run the train-loop.

Any help would be much appreciated!

**Comment by Atharva-Phatak** [comment]
I took a look at your notebook, your input images do not seem to be RGB. I think you will need to change to input structure of the model( the first conv layer). Your dataloader seems to be fine. 

I recommend you do a single pass with the help of your model
i.e do a pass with single image and its corresponding targets,

Also if you can post the error report here and shapes of your input image, I think I will be able to help you.
